### PR TITLE
feat: Change uploading file output

### LINKED ISSUE #1334: Check and improve uploading file console output
**Author the user story for this feature**

As a User, I want to improve output for uploading files so I can easily see which files are uploaded.

**Describe the solution**
[Based on discussion in PR](https://github.com/Flank/flank/pull/1329#pullrequestreview-533495566)

Let's decide what we want to print actually when we uploading files to gcloud. 
Currently, we just print the message 
`Uploading [file name] `
do we want to stick with that? Or we can print a gcs link but on a matrix level? So every matrix will have it's own. IMO we should decide on one convention and not mix them

### PR DESCRIPTION
**Pull request by piotradamczyk5**
Fixes #1334 

Previous uploading Flank output looks like:
```
Saved 1 shards to android_shards.json
  Uploading android_shards.json .
  Uploading app-debug.apk .
  Uploading app-debug-androidTest.apk .
  1 test / 1 shard

  1 matrix ids created in 0m 3s
  https://console.developers.google.com/storage/browser/test-lab-v9cn46bb990nx-kz69ymd4nm9aq/test_dir

```

The new output for file upload looks more like Gcloud output:
```
  Saved 1 shards to android_shards.json
  Uploading [android_shards.json] to https://console.developers.google.com/storage/browser/test-lab-v9cn46bb990nx-kz69ymd4nm9aq/test_dir/...
  Uploading [app-debug.apk] to https://console.developers.google.com/storage/browser/test-lab-v9cn46bb990nx-kz69ymd4nm9aq/test_dir/...
  Uploading [app-debug-androidTest.apk] to https://console.developers.google.com/storage/browser/test-lab-v9cn46bb990nx-kz69ymd4nm9aq/test_dir/...

  1 test / 1 shard

  1 matrix ids created in 0m 3s
  Raw results will be stored in your GCS bucket at [https://console.developers.google.com/storage/browser/test-lab-v9cn46bb990nx-kz69ymd4nm9aq/test_dir]
```

Gcloud file upload output for reference:
```
Uploading [test_runner/../test_projects/android/apks/app-debug.apk] to Firebase Test Lab...
Uploading [test_runner/../test_projects/android/apks/app-debug-androidTest.apk] to Firebase Test Lab...
Raw results will be stored in your GCS bucket at [https://console.developers.google.com/storage/browser/test-lab-v9cn46bb990nx-kz69ymd4nm9aq/test_dir/]

```

1. Run Flank
1. Check output
1. Output for file upload looks like described above

## Checklist

- [x] Research done
- [x] Some refactor of output
- [x] Unit tested
- [x] Integration tests updated

### CODE REVIEW

**Comment by piotradamczyk5** [comment]
@flank-it

**Comment by piotradamczyk5** [comment]
@flank-it

File: test_runner/src/test/kotlin/ftl/run/GenericTestRunnerTest.kt [hunk 1]
```diff
@@ -31,7 +33,7 @@ class GenericTestRunnerTest {
         val result = createMock(1).beforeRunMessage(
             listOf(Chunk(listOf(TestMethod(name = "", time = 0.0))))
         ).normalizeLineEnding()
-        assert(result, "  1 test / 1 shard\n")
+        assert(result, "  1 test / 1 shard".withNewLineAtTheEnd().startWithNewLine())
```
Comment (Sloox) [inline test_runner/src/test/kotlin/ftl/run/GenericTestRunnerTest.kt:36]: can not the same be achieved with
```suggestion
        assert(result.trim(), "  1 test / 1 shard".trim())
```

File: test_runner/src/test/kotlin/ftl/util/ProgressBarTest.kt [hunk 2]
```diff
@@ -49,11 +49,12 @@ class ProgressBarTest {
         runWithProgress(
             startMessage = "TEST",
             action = { action.run() },
+            onSuccess = {},
```
Comment (Sloox) [inline test_runner/src/test/kotlin/ftl/util/ProgressBarTest.kt:52]: We dont need teh onSuccess as it has a default?

File: test_runner/src/test/kotlin/ftl/util/ProgressBarTest.kt [hunk 3]
```diff
@@ -75,11 +76,12 @@ class ProgressBarTest {
                 action.run()
                 throw Exception()
             },
+            onSuccess = {},
```
Comment (Sloox) [inline test_runner/src/test/kotlin/ftl/util/ProgressBarTest.kt:79]: same here

**Review by Sloox — CHANGES_REQUESTED** [review]

File: test_runner/src/test/kotlin/ftl/util/ProgressBarTest.kt (hunk 2 above)
Comment (piotradamczyk5) [reply test_runner/src/test/kotlin/ftl/util/ProgressBarTest.kt:52]: fixed

File: test_runner/src/test/kotlin/ftl/util/ProgressBarTest.kt (hunk 3 above)
Comment (piotradamczyk5) [reply test_runner/src/test/kotlin/ftl/util/ProgressBarTest.kt:79]: fixed

File: test_runner/src/test/kotlin/ftl/run/GenericTestRunnerTest.kt (hunk 1 above)
Comment (piotradamczyk5) [reply test_runner/src/test/kotlin/ftl/run/GenericTestRunnerTest.kt:36]: replaced

**Comment by adamfilipow92** [comment]
@flank-it

**Comment by piotradamczyk5** [comment]
> **Integration tests failed ❌**
> Build scan https://gradle.com/s/cztzvgnxnwwui
> Workflow run https://github.com/Flank/flank/actions/runs/452641114

Quote exceed 😞

File: integration_tests/src/test/kotlin/integration/AllTestFilteredIT.kt [hunk 4]
```diff
@@ -2,6 +2,7 @@ package integration
 
 import FlankCommand
 import com.google.common.truth.Truth.assertThat
+import flank.common.isWindows
```
Comment (Sloox) [inline integration_tests/src/test/kotlin/integration/AllTestFilteredIT.kt:5]: I am so glad we have a commons now 👍

**Review by Sloox — APPROVED** [review]

**Review by pawelpasterz — CHANGES_REQUESTED** [review]
I run locally and have one test failed  (except the other two failed due to quota exceeded.... 😓 )
Of course, I forgot to use `--scan` flag so the only thing I can share is the gist from output https://gist.github.com/pawelpasterz/52b4ce341265e3b982607992b67a3dc3

**Comment by piotradamczyk5** [comment]
@flank-it

**Review by pawelpasterz — APPROVED** [review]

**Pull request merged**